daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -434,144 +526,9 @@ private[lf] class Runner(
-                              Future.failed(exce)
-                            case SOptional(Some(free)) =>
-                              Future.successful(SEApp(SEValue(continue), Array(free)))
-                            case e =>
-                              Future.failed(
-                                new ConverterException(s"Expected SOptional but got $e")
-                              )
-                          }
-                      case Failure(e) => Future.failed(e)
-                    }
-                  case ScriptF.Throw(SAny(ty, value)) =>
-                    Future.failed(
-                      Runner.InterpretationError(
-                        SError
-                          .SErrorDamlException(IE.UnhandledException(ty, value.toUnnormalizedValue))
-                      )
-                    )
-                  case cmd: ScriptF.Cmd =>
-                    cmd.execute(env).transform {
-                      case f @ Failure(Runner.CanceledByRequest) => f
-                      case f @ Failure(Runner.TimedOut) => f
-                      case Failure(exception) =>
-                        Failure(new ScriptF.FailedCmd(cmd, exception))
-                      case Success(value) =>
-                        Success(value)
-                    }
-                }
-              }
-              .flatMap(run(_))
-          case Left(v) =>
-            v match {
-              case SRecord(_, _, ArrayList(newState, _)) => {
-                // Unwrap the Tuple2 we get from the inlined StateT.
-                Future { newState }
-              }
-              case _ => Future.failed(new ConverterException(s"Expected Tuple2 but got $v"))
-            }
-        }
-    }
-
-    val resultF = for {
-      _ <- Future.unit // We want the evaluation of following stepValue() to happen in a future.
-      result <- stepToValue().fold(Future.failed, Future.successful)
-      expr <- result match {
-        // Unwrap Script type and apply to ()
-        // For backwards-compatibility we support the 1 and the 2-field versions.
-        case SRecord(_, _, vals) if vals.size == 1 || vals.size == 2 => {
-          vals.get(0) match {
-            case SPAP(_, _, _) =>
-              Future(SEAppAtomic(SEValue(vals.get(0)), Array(SEValue(SUnit))))
-            case _ =>
-              Future.failed(
-                new ConverterException(
-                  "Mismatch in structure of Script type. " +
-                    "This probably means that you tried to run a script built against an " +
-                    "SDK <= 0.13.55-snapshot.20200304.3329.6a1c75cf with a script runner from a newer SDK."
-                )
-              )
-          }
-        }
-        case v => Future.failed(new ConverterException(s"Expected record with 1 field but got $v"))
-      }
-      v <- run(expr)
-    } yield v
-    (machine, resultF)
+    new v1.Runner(this).runWithClients(initialClients, traceLog, warningLog, canceled)
```

Comment:
Defer running logic to versioned runner. v1 running logic in v1.Runner is unchanged.